android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt

```diff
@@ -143,4 +154,65 @@ class ProfileViewModelTest : RobolectricTest() {
     )
     coVerify { registerRepository.changeManagingEntity(any(), any(), any()) }
   }
+
+  @Test
+  fun testOverflowMenuClickChangeManagingEntity() {
+    val mockEvent = mockk<ProfileEvent.OverflowMenuClick>()
+    val mockActionConfig = mockk<ActionConfig>()
+    val listActionConfig = mockk<List<ActionConfig>>()
+    val mockWorkflow = ApplicationWorkflow.CHANGE_MANAGING_ENTITY
+    val mockManagingEntity = mockk<ManagingEntityConfig>()
+
+    every { mockEvent.overflowMenuItemConfig?.actions } returns listActionConfig
+    every { mockActionConfig.managingEntity } returns mockManagingEntity
+    every { mockActionConfig.interpolate(any()) } returns mockActionConfig
+    every { mockActionConfig.workflow } returns mockWorkflow.name
+    every { mockActionConfig.managingEntity } returns mockManagingEntity
+
+    val changeManagingEntity =
+      profileViewModel.javaClass.getDeclaredMethod(
+        "changeManagingEntity",
+        ProfileEvent.OverflowMenuClick::class.java,
+        ManagingEntityConfig::class.java,
+      )
```

Comment:
Can we change the scope of this function to `public` so we do not have to use reflection to call the function via reflection? It is okay to expose ViewModel functions for testing and to their callers.